Sources/SourceControl/Repository.swift

```diff
@@ -118,12 +118,23 @@ public protocol RepositoryProvider {
     ///   - path: The location of the repository on disk, at which the repository
     ///     has previously been created via `cloneCheckout`.
     func openCheckout(at path: AbsolutePath) throws -> WorkingCheckout
+
+    /// Copies the repository at path `from` to path `to`.
+    /// - Parameters:
+    ///   - sourcePath: the source path.
+    ///   - destinationPath: the destination  path.
+    func copy(from sourcePath: AbsolutePath, to destinationPath: AbsolutePath) throws
 }
 
 extension RepositoryProvider {
     public func checkoutExists(at path: AbsolutePath) throws -> Bool {
         fatalError("Unimplemented")
     }
+
+    public func copy(from sourcePath: AbsolutePath, to destinationPath: AbsolutePath) throws {
+        fatalError("Unimplemented")
```

Comment:
This is only called if caching is enabled, right?  So that existing clients would only get this fatal error if they enable caching?

Reply:
This is only called once at RepositoryManager:369 and only if a cachePath is supplied.

Reply:
Great, thanks!